File: java/muse/algorithms/search/LinearSearch.java
package muse.algorithms.search;

public final class LinearSearch {

  private LinearSearch() {}

  public static <T extends Comparable<T>> int find(T[] arr, T key) {
    for (int i = 0, length = arr.length; i < length; i++) {
      if (arr[i].compareTo(key) == 0) {
        return i;
      }
    }

    return arr.length;
  }
}

File: archive/php/muse/algorithms/search/LinearSearch.php
<?php

declare(strict_types=1);

namespace muse\algorithms\search\LinearSearch;

function find(array &$arr, $key): int
{
    foreach ($arr as $i => $v) {
        if ($v == $key) {
            return $i;
        }
    }

    return count($arr);
}
